examples/learning/learn_associations.ipynb

```diff
@@ -0,0 +1,330 @@
+     "source": [
+      "## Step 2: Build the model\n",
+      "\n",
+      "We define a helper function that is useful for creating nodes that cycle through keys/values."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "def cycle_array(x, period, dt=0.001):\n",
+      "    \"\"\"Cycles through the elements\"\"\"\n",
+      "    i_every = int(round(period/dt))\n",
+      "    if i_every != period/dt:\n",
+      "        raise ValueError(\"dt (%s) does not divide period (%s)\" % (dt, period))\n",
+      "    def f(t):\n",
+      "        i = int(round((t - dt)/dt))  # t starts at dt\n",
+      "        return x[(i/i_every)%len(x)]\n",
+      "    return f"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": [],
+     "prompt_number": null
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "We create three inputs: the keys, the values, and a modulatory learning signal. The model is run continuously in two phases: the first half learns the set of associations, and the second tests recall.\n",
+      "\n",
+      "The learning signal will be set to 0 to allow learning during the first phase, and -1 to inhibit learning during the second phase.\n",
+      "\n",
+      "The memory is confined to a single ensemble. Roughly speaking, its encoders will hold the keys, and its decoders will hold the values. "
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "# Model constants\n",
+      "n_neurons = 200\n",
+      "dt = 0.001\n",
+      "period = 0.3\n",
+      "T = period*num_items*2\n",
+      "\n",
+      "# Model network\n",
+      "model = nengo.Network()\n",
+      "with model:\n",
+      "    \n",
+      "    # Create the inputs/outputs\n",
+      "    stim_keys = nengo.Node(output=cycle_array(keys, period, dt))\n",
+      "    stim_values = nengo.Node(output=cycle_array(values, period, dt))\n",
+      "    learning = nengo.Node(output=lambda t: -int(t>=T/2))\n",
+      "    recall = nengo.Node(size_in=d_value)\n",
+      "    \n",
+      "    # Create the memory\n",
+      "    memory = nengo.Ensemble(n_neurons, d_key, intercepts=[intercept]*n_neurons)\n",
+      "    \n",
+      "    # Learn the encoders/keys\n",
+      "    voja = nengo.Voja(post_tau=None, learning_rate=5e-2)\n",
+      "    conn_in = nengo.Connection(stim_keys, memory, synapse=None,\n",
+      "                               learning_rule_type=voja)\n",
+      "    nengo.Connection(learning, conn_in.learning_rule, synapse=None)\n",
+      "    \n",
+      "    # Learn the decoders/values, initialized to a null function\n",
+      "    conn_out = nengo.Connection(memory, recall, learning_rule_type=nengo.PES(1e-3),\n",
+      "                                function=lambda x: np.zeros(d_value))\n",
+      "    \n",
+      "    # Give the error connection positive intercepts so that it can be inhibited\n",
```

Comment:
Right, I forgot about that! Though in this case we're using a weight of `10`, so it will definitely be below any intercepts, so I still think it's a bit confusing / unnecessary to set the intercepts.